clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/ResolveTitusJobId.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.titus.deploy.actions;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.netflix.spinnaker.clouddriver.event.EventMetadata;
+import com.netflix.spinnaker.clouddriver.saga.SagaCommand;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.models.Saga;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.titus.TitusClientProvider;
+import com.netflix.spinnaker.clouddriver.titus.client.TitusClient;
+import com.netflix.spinnaker.clouddriver.titus.client.model.Job;
+import com.netflix.spinnaker.clouddriver.titus.credentials.NetflixTitusCredentials;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.DestroyTitusJobDescription;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.DestroyTitusServerGroupDescription;
+import java.util.Collections;
+import javax.annotation.Nonnull;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ResolveTitusJobId implements SagaAction<ResolveTitusJobId.ResolveTitusJobIdCommand> {
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final TitusClientProvider titusClientProvider;
+
+  @Autowired
+  public ResolveTitusJobId(
+      AccountCredentialsProvider accountCredentialsProvider,
+      TitusClientProvider titusClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.titusClientProvider = titusClientProvider;
+  }
+
+  @NotNull
+  @Override
+  public Result apply(
+      @NotNull ResolveTitusJobId.ResolveTitusJobIdCommand command, @NotNull Saga saga) {
+    AccountCredentials accountCredentials =
+        accountCredentialsProvider.getCredentials(command.description.getAccount());
+
+    TitusClient titusClient =
+        titusClientProvider.getTitusClient(
+            (NetflixTitusCredentials) accountCredentials, command.description.getRegion());
+
+    Job job = titusClient.findJobByName(command.description.getServerGroupName());
+    if (job != null) {
+      DestroyTitusJobDescription destroyTitusJobDescription = new DestroyTitusJobDescription();
```

Comment:
This worked first shot ... is it legitimate? 

@robzienert Is this what you were referring to with context needing to be explicitly passed forward?

Reply:
> .. is it legitimate?

Which part, creating a description inside of the action? Yeah, that's legit.

> @robzienert Is this what you were referring to with context needing to be explicitly passed forward?

Sort of. This is actually what we want, actions that can be composed together. The thing where the design is flawed is in the `Result` object, requiring a `Command` for the next action to run - this makes each action tightly coupled, which isn't desired.

We can change things such that actions return _only_ events, and a command is only used for kicking off a Saga. Actions, then, would just specify the events they require, and those events would be injected in `apply()` instead of a `Command`.

So, in this case instead of returning a `Result(DestroyTitusJobCommand)` we'd return a `Result(TitusJobIdResolved)` and then `DestroyTitusJobAction` would require `TitusJobIdResolved`. The only thing linking the two actions together would be the Flow.

Does that make sense?

Reply:
Looks legit to me. Though I suppose the description could also just be passed forward.